documentation/seed-node.md

```diff
@@ -0,0 +1,255 @@
+
+`subscription.backfillOnly` will tell the seed-node to only send historical data and not stream incoming data going forward.
+
+`subscription.startingBlock` is the starting block number for the range we want to receive data in.
+
+`subscription.endingBlock` is the ending block number for the range we want to receive data in;
+setting to 0 means there is no end/we will continue indefinitely.
+
+`subscription.headerFilter` has two sub-options: `off` and `finalOnly`. Setting `off` to true tells the seed-node to
+not send any headers to the subscriber; setting `finalOnly` to true tells the seed-node to send only canonical headers.
+
+`subscription.trxFilter` has three sub-options: `off`, `src`, and `dst`. Setting `off` to true tells the seed-node to
+not send any transactions to the subscriber; `src` and `dst` are string arrays which can be filled with ETH addresses we want to filter transactions for,
+if they have any addresses then the seed-node will only send transactions that were sent or received by the addresses contained
+in `src` and `dst`, respectively.
+
+`subscription.receiptFilter` has two sub-options: `off` and `topics`. Setting `off` to true tells the seed-node to
+not send any receipts to the subscriber; `topic0s` is a string array which can be filled with event topics we want to filter for,
+if it has any topics then the seed-node will only send receipts that contain logs which have that topic0.
+
+`subscription.stateFilter` has three sub-options: `off`, `addresses`, and `intermediateNodes`. Setting `off` to true tells the seed-node to
+not send any state data to the subscriber; `addresses` is a string array which can be filled with ETH addresses we want to filter state for,
+if it has any addresses then the seed-node will only send state leafs (accounts) corresponding to those account addresses. By default the seed-node
+only sends along state leafs, if we want to receive branch and extension nodes as well `intermediateNodes` can be set to `true`.
+
+`subscription.storageFilter` has four sub-options: `off`, `addresses`, `storageKeys`, and `intermediateNodes`. Setting `off` to true tells the seed-node to
+not send any storage data to the subscriber; `addresses` is a string array which can be filled with ETH addresses we want to filter storage for,
+if it has any addresses then the seed-node will only send storage nodes from the storage tries at those state addresses. `storageKeys` is another string
+array that can be filled with storage keys we want to filter storage data for. It is important to note that the storageKeys are the actual keccak256 hashes, whereas
+the addresses in the `addresses` fields are the ETH addresses and not their keccak256 hashes that serve as the actual state keys. By default the seed-node
+only sends along storage leafs, if we want to receive branch and extension nodes as well `intermediateNodes` can be set to `true`.
+
```

Comment:
This documentation is awesome! I wonder at some point if we want to include a brief description of the seed node in the main README as well. Totally doesn't have to be part of this PR though.

Reply:
I may need to read this more closely, but I'm not sure I entirely understand the usecase of the `syncAndPublish` command. Would this be for users that cared about getting statediffs and/or block data off of IPFS specifically? And then for a something like `mcd_transformers`, we'd likely use `syncPublishScreenAndServe` I think? 🤔 

This is kind of me just thinking "aloud", so feel free to skip this comment/question for now. 😄

Reply:
I separated syncAndPublish out into its own process/command separate from the "screenAndServe" processes because I thought there _might_ be a case where we want to seed data but are not yet interested in exposing/serving the data to transformers.

Reply:
This is absolutely the case. For example, the seed node will be loading all of Ethereum into IPFS before any transformers connect to it.

Reply:
Doesn't it make sense to separate these two modes of operation completely into different commands? Like we do with `headerSync` and `composeAndExecute` for example.

Reply:
`screenAndServe` would be limited to only screening and serving previously synced and published data if there is not a `syncAndPublish` process going on at the same time, but I agree that otherwise it would make sense to split them up this way.

Reply:
Yeah, in the same way the application works when running `composeAndExecute` without `headerSync`?

Reply:
Yeah that's a good analogy, this can already be done with the current methods although I haven't written a command for it.